prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -118,19 +123,32 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
                         )
                         logger.info("Token successfully added!")
 
+                        // If encryption is needed.
+                        reportContent = if (restTransportInfo.encryptionKeyUrl.isNotEmpty()) {
+                            encryptTheReport(
+                                reportContent,
+                                restTransportInfo.encryptionKeyUrl,
+                                httpHeaders,
+                                httpClient ?: createDefaultHttpClient(
+                                    jksCredential, accessToken,
+                                    restTransportInfo
+                                )
+                            )
+                        } else {
+                            reportContent
+                        }
```

Comment:
You can remove the else if you do it this way:
```
// If encryption is needed.
 if (restTransportInfo.encryptionKeyUrl.isNotEmpty()) {
     reportContent =   encryptTheReport(
        reportContent,
        restTransportInfo.encryptionKeyUrl,
        httpHeaders,
        httpClient ?: createDefaultHttpClient(
            jksCredential, accessToken,
            restTransportInfo
        )
    )
}
```

Reply:
Oh, Yes.  Thanks